python/revng/api/_capi.py

```diff
@@ -3,13 +3,62 @@
 #
 
 import os
+import re
 import traceback
+from functools import wraps
 from pathlib import Path
-from typing import List
+from typing import Any, Callable, Dict, List
 
 from cffi import FFI
 from cffi.backend_ctypes import CTypesBackend
 
+
+class ApiWrapper:
+    function_matcher = re.compile(
+        r"(?P<return_type>[\w_]+)\s*\*\s*\/\*\s*owning\s*\*\/\s*(?P<function_name>[\w_]+)",
+        re.M | re.S,
+    )
```

Comment:
@mrjackv does this work with multiline prototype declarations too?

Reply:
Yes, `re.M` is for multiline